backend/src/main/java/com/votogether/domain/vote/service/VoteService.java

```diff
@@ -0,0 +1,59 @@
+package com.votogether.domain.vote.service;
+
+import com.votogether.domain.member.entity.Member;
+import com.votogether.domain.post.entity.Post;
+import com.votogether.domain.post.entity.PostOption;
+import com.votogether.domain.post.repository.PostRepository;
+import com.votogether.domain.vote.entity.Vote;
+import com.votogether.domain.vote.repository.VoteRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class VoteService {
+
+    public final VoteRepository voteRepository;
+    public final PostRepository postRepository;
+
+    public void vote(
+            final Member member,
+            final Long postId,
+            final Long postOptionId
+    ) {
+        Post post = postRepository.findById(postId)
+                .orElseThrow(() -> new IllegalArgumentException("게시글이 존재하지 않습니다."));
+        validateDeadLine(post);
+
+        Vote vote = member.vote(post, postOptionId);
+        member.plusPoint(1);
+
+        voteRepository.save(vote);
+    }
+
+    private void validateDeadLine(final Post post) {
+        if (post.isClosed()) {
+            throw new IllegalStateException("게시글이 마감되었습니다.");
+        }
+    }
+
+    public void changeVote(
+            final Member member,
+            final Long postId,
+            final Long originPostOptionId,
+            final Long newPostOptionId
+    ) {
+        Post post = postRepository.findById(postId)
+                .orElseThrow(() -> new IllegalArgumentException("게시글이 존재하지 않습니다."));
+        validateDeadLine(post);
+
+        Vote vote = voteRepository.findByMemberIdAndPostOptionId(
```

Comment:
기존에 투표한 옵션이 존재하지 않으면 예외 상황으로 볼 수 있지 않을까요?!

Reply:
```suggestion
        Vote vote = voteRepository.findByMemberAndPostOption(
```
`Member`, `PostOption` 2개 다 찾은 상태이니 객체를 통해 찾는 방법은 어떨까요? ID로 조회할 때와 객체로 조회할 때는 어떤 차이가 있을까요?

Reply:
레파지토리가 컬렉션의 역할을 하는만큼 객체로 파라미터를 받게하니 더 객체지향스러워졌어요~ 👍